content/2017.03.building-projects-as-a-distributed-team.rst

```diff
@@ -0,0 +1,178 @@
+
+Write a document (article, blue print) to explain what you decided to
+build as a prototype.
+
+At this stage it is really important to get something that looks close
+enough to the minimum feature set we want, but it should be really
+simple. In two days you should be able to demo the UX of your
+prototype and in two weeks something working well enough.
+
+It is really important to do that because it will bring a lot of
+questions, you will realize that the prototype is different from each
+thoughts of how it should look likes.
+
+This is especially true if you are working with people from differents
+jobs that are not looking at the same things. (Backend developers, UX
+designers, users, ...)
+
+The faster you get your prototype ready the faster you can align on
+what you all want for your MVP.
+
+
+The Minimal Viable Product (MVP)
+================================
+
+Now that we have a prototype, the question now is what is missing to
+make a product.
+
+It is time to gather together and brainstorm around the
+project to define what is the **Minimum Viable Product**.
+
+At this stage it is really important to start talking of a product,
+the project version ``1.0.0`` starts with a ``1`` which means that you
+are trying to put together the first actual version of your product.
+
+Everything that is not in the feature set of the MVP is put in a
+backlog for later.
+
+It is important to document the fact that we thought about it and that
+we will possibly care about it later.
+
+At each stage — Prototype, MVP — we write a small article to summarize
+the discussion and to explain what is the feature set, what is the
+protocol.
+
+You might think that writing an article about it should be optional,
+but you will realize that if you don't do it, you won't remember on
+what you'd agree and people of your team will start asking questions
+about why you did it like that. If the expectation change along the
+way we should notice it and take decision to update the document.
+
+
+Answering complex questions
+---------------------------
+
+Some features of the MVP might not be trivial. Some decision are hard
+to make between two solutions.
+
+You will have to carefully explain pros and cons and why you choose to
+do it like that. If it is written it is easy to explain people who
+forgot about it why you did it like that.
+
+We have a bunch of examples of hard problems:
+
+- How do you handle permissions in Kinto?
+- How do we support multiple database backends?
+- How do we handle ``content-editable`` divs with React or Elm?
+- How do you implement an OAuth flow from a web extension or a Mobile app?
+- How do you handle pluggable notifications in Kinto?
+- How do we configure email notifications in Kinto?
+- How do we let people attach files to their records in Kinto?
+- How do we get Firefox Account keys derived from the password from
+  an OAuth flow when we never get our hands on the password?
```

Comment:
Because the previous paragraphs were about having to make a difficult choice between two solutions, I expected "hard problems" to be about those kinds of problems. But mostly these don't seem like those kinds of problems. Maybe you need a transition before `hard problems`? Maybe we shouldn't call them `hard` but something else?